inventory_management_system_api/services/item.py

```diff
@@ -118,9 +121,75 @@ def get(self, item_id: str) -> Optional[ItemOut]:
         :param item_id: The ID of the item to retrieve
         :return: The retrieved item, or `None` if not found
         """
-
         return self._item_repository.get(item_id)
 
+    def update(self, item_id: str, item: ItemPatchRequestSchema) -> ItemOut:
+        """
+        Update an item by its ID.
+
+        The method checks if the item exists in the database and raises a `MissingRecordError` if it does
+        not. If the catalogue item ID or system ID is being updated, it checks if a catalogue item
+        and/or system ID with such ID exists and raises a `MissingRecordError` if it does not.
+        When updatimg properties, existing properties must all be supplied, or they will be overwritten
+        by the catalogue item properties.
+
+        :param item_id: The ID of the item to update.
+        :param item: The item containing the fields that need to be updated.
+        :return: The updated item.
+        """
+        update_data = item.model_dump(exclude_unset=True)
+
+        stored_item = self.get(item_id)
+        if not stored_item:
+            raise MissingRecordError(f"No item found with ID: {item_id}")
+
+        catalogue_item = None
+        if "catalogue_item_id" in update_data and item.catalogue_item_id != stored_item.catalogue_item_id:
+            catalogue_item = self._catalogue_item_repository.get(item.catalogue_item_id)
+            if not catalogue_item:
+                raise MissingRecordError(f"No catalogue item found with ID: {item.catalogue_item_id}")
+
+            if "properties" not in update_data:
+                item.properties = [PropertyPostRequestSchema(**prop.model_dump()) for prop in stored_item.properties]
+                update_data = item.model_dump(exclude_unset=True)
```

Comment:
It was confirmed yesterday that this case won't occur so we should not allow for an item to be moved from one catalogue item to another. I wonder if it is possible in pydantic to remove the inherited `catalogue_item_id` property from the `ItemPatchRequestSchema` model. If that's possible then that can be one solution. Another solution would be to raise an exception here. @joelvdavies thoughts?

Once you address this, please remove or update any tests that test this logic.

Reply:
Ah sorry only just saw this comment, I prefer the current solution though - raising an error as I think removing inherited variables could be a bit confusing and I don't think it is possible without using another base class.